### ISSUE TITLE
False positive about unused variable when using C#'s 8 new using directive

### ISSUE THREAD

**Issue by GeeWee** [SonarSource/sonar-dotnet#3492]
### Description

Rule S1481 has a false positive on unused local variables, which are in a using statement.

E.g. the following line 
```csharp
using var _ = LogContext.PushProperty("foo", "foo");
// other stuff here, _ is never used
```
(using LogContext in Serilog adds additional context to all logs in scope of it. The only reason to use the return value is to dispose of it afterwards)

It's the same as the example in the description that is specifically stated should not trigger it:
```C#
using(var t = new TestTimer()) // t never used, but compliant.
{
  //...
}
```

### Related information

* C#/VB.NET Plugins Version. Using SonarQube 8.4 community via Docker
* Visual Studio Version: N/A (using Rider)
* MSBuild / dotnet version: 3.1.106
* If running through the Scanner for MSBuild, its version. `SonarScanner for MSBuild 4.10`
* Operating System: Fedora 32

**Comment by pavel-mikula-sonarsource** [comment]
Hi @GeeWee,

S3604 is about redundant member initialization inside a class and that doesn't seem relevant to your code examples. Did you mean S1481?

**Comment by GeeWee** [comment]
I did! I must've done a copy-paste good. I've updated the original issue.

**Comment by pavel-mikula-sonarsource** [comment]
Thanks for the clarification and reporting this case. 

We already have an issue #3137 in the backlog. It has the same scenario so you can follow that one. I'll close this report.

**Comment by GeeWee** [comment]
Yes, I just missed it. Thanks!
